### ISSUE TITLE
Can't get it working for ESPEasy 2.0 and ESP8285

### ISSUE THREAD

**Issue by lhurt** [soif/EspBuddy#15]
Hi,
I love the idea of your wrapper and would like to use them on my Sonoff Touch switches which are based on ESP8285. Due to the size of the firmware I have to use the two step approach.

Unfortunately even the first upload step fails. The concerning device runs ESPEasy 2.0.0-dev12.

The image compiles after removing -DUSE_CUSTOM_H from the options. Otherwise the build fails because I have no custom parts.
The wizard finds my device and I can retrieve information from it, like version.
But the upload fails with the following lines
```
** Uploading Intermediate Uploader Firmware ****************************************************************************
-> /opt/EspBuddy/bin/espota.py -r -d -i 192.168.1.61  -f /opt/EspBuddy/firmwares/ESPEasyUploaderMega.OTA.1m128.bin
08:30:36 [DEBUG]: Options: {'esp_ip': '192.168.1.61', 'host_port': 47070, 'image': '/opt/EspBuddy/firmwares/ESPEasyUploaderMega.OTA.1m128.bin', 'host_ip': '0.0.0.0', 'auth': '', 'esp_port': 8266, 'spiffs': False, 'debug': True, 'progress': True}
08:30:36 [INFO]: Starting on 0.0.0.0:47070
08:30:36 [INFO]: Upload size: 288064
08:30:36 [INFO]: Sending invitation to: 192.168.1.61
08:30:46 [ERROR]: No Answer
```

I've tried to start this manually and telnet connects on port 47070.

Any idea?

Here are the relevant parts of my config.php
```
$cfg['repos']['espeasy']['path_repo']                           ="/opt/ESPEasy/";

$my_espeasy_flags ='-DBUILD_DEV=\"'.$my_build.'\" -DMY_IP=\"{{host_ip}}\" -DMY_AP_IP={{host_ip1}},{{host_ip2}},{{host_ip3}},{{host_ip4}} -DMY_NAME=\"{{host_name}}\" -DMY_UNIT={{host_ip4}}';

$cfg['configs']['espeasy_1024_8285']['repo']                                                    ="espeasy";
$cfg['configs']['espeasy_1024_8285']['environment']                                             ="normal_ESP8285_1024";
$cfg['configs']['espeasy_1024_8285']['exports']['PLATFORMIO_BUILD_FLAGS']                       =$my_espeasy_flags;
$cfg['configs']['espeasy_1024_8285']['2steps']                                                  =true;

$cfg['hosts']['sonoff_touch']['ip']             ="192.168.1.61";
$cfg['hosts']['sonoff_touch']['config']         ="espeasy_1024_8285";
```

Any help is appreciated

**Comment by soif** [comment]
I'm currently working on a fix for one of my small 1M devices. 
I'm faced with a similar issue that prevents OTA to work on latest ESPEasy dev firmwares.

To verify if it is the same issue, try to monitor your ESP on SerialPort 
`espbuddy.php monitor sonoff_touch espeasy_1024_8285 -v --rate=115200 --port=/dev/xxx`

while starting an OTA upload on another console:
`espbuddy.php upload sonoff_touch espeasy_1024_8285 -vb`

If this is the same issue, you should see a message saying _OTA  : Error (will reboot): Begin Failed_

-----
BTW in order to get OTA working, your current running firmware MUST have be compiled with the **FEATURE_ARDUINO_OTA** defined, ie:

`$my_espeasy_flags=''-DFEATURE_ARDUINO_OTA;` or de-commented in the ESPEASY.ino file.

You can verify on the boot log in the serial console that you really have a message saying: _OTA  : Arduino OTA enabled on port 8266_

**Comment by lhurt** [comment]
Hi,
thanks for the reply. As I flashed the precompiled firmware on the Touch device I assume that the option was disabled.

Btw. How to I monitor the serial port (on Windows)?
Do I have to wire the ESP to the USB TTL converter?
I never did that before.

Thanks

**Comment by soif** [comment]
>As I flashed the precompiled firmware on the Touch device I assume that the option was disabled

Yes Option is disabled by default

> How to I monitor the serial port (on Windows)? Do I have to wire the ESP to the USB TTL converter?

yep or you can monitor serial directly on the web , or on a syslog server (see ESPeasy avanced setting about this)

**Comment by lhurt** [comment]
Hi,
now I'm pretty confused. I recompiled ESPEasy with OTA enabled. On reboot the console reads "OTA  : Arduino OTA enabled on port 8266". But telnet to this port results in "connection refused" and manual execution of espota.py says "No response from device".

Any idea what to do?

**Comment by soif** [comment]
you certainly need to provide the login/password

**Comment by lhurt** [comment]
I didn't set any

**Comment by soif** [comment]
if i remember correly this is mandatory on first espurna boot, so U must have set up one, then use it in the configuration.

**Comment by lhurt** [comment]
I don't use espurna. I'm using ESPEasy.

**Comment by soif** [comment]
sorry my bad

**Comment by lhurt** [comment]
I'll recompile and test again. May be I did something wrong.
I used the EspBuddy build script. May be I should use Arduino IDE,

**Comment by soif** [comment]
this may be because the firmware you try to upload is too large:

Connect to the Web GUI of the intermediate firmware (if it has been succefully uploaded), read the max size for flash, then compare it to the firmware size. if firmware is larger, espota.py fails with a message that dont explain that at all..

BTW use EspBuddy to upload, not EspOta.py directly

**Comment by soif** [comment]
Also be sure to use the LATEST EspBuddy, and take a look at the new config-sample

**Comment by lhurt** [comment]
OK,
one step forward.

After recompiling the dev-12 firmware with OTA enabled again and flashing it to my Sonoff Touch the upload of the intermediate image was successful.

The relevant output of EspBuddy is
```
** Uploading Intermediate Uploader Firmware ****************************************************************************
-> /opt/EspBuddy/bin/espota.py -r -d -i 192.168.1.61  -f "/opt/EspBuddy/firmwares/ESPEasyUploaderMega.OTA.1m128.bin"
21:06:43 [DEBUG]: Options: {'esp_ip': '192.168.1.61', 'host_port': 33502, 'image': '/opt/EspBuddy/firmwares/ESPEasyUploaderMega.OTA.1m128.bin', 'host_ip': '0.0.0.0', 'auth': '', 'esp_port': 8266, 'spiffs': False, 'debug': True, 'progress': True}
21:06:43 [INFO]: Starting on 0.0.0.0:33502
21:06:43 [INFO]: Upload size: 288064
21:06:43 [INFO]: Sending invitation to: 192.168.1.61
21:06:43 [INFO]: Waiting for device...
Uploading: [============================================================] 100% Done...

21:06:52 [INFO]: Waiting for result...
21:06:52 [INFO]: Result: OK


** Waiting  16 sec for ESP to reboot 16 15 14 13 12 11 10 9 8 7 6 5 4 3 2 1  ********


** Uploading Final Firmware ********************************************************************************************
-> /opt/EspBuddy/bin/espota.py -r -d -i 192.168.1.61  -f "/tmp/EspBuddy/wc-schalter-160.fritz.box/Firmware-espeasy_1024_8285-v20100.bin"
21:07:08 [DEBUG]: Options: {'esp_ip': '192.168.1.61', 'host_port': 13149, 'image': '/tmp/EspBuddy/wc-schalter-160.fritz.box/Firmware-espeasy_1024_8285-v20100.bin', 'host_ip': '0.0.0.0', 'auth': '', 'esp_port': 8266, 'spiffs': False, 'debug': True, 'progress': True}
21:07:08 [INFO]: Starting on 0.0.0.0:13149
21:07:08 [INFO]: Upload size: 607856
21:07:08 [INFO]: Sending invitation to: 192.168.1.61
21:07:18 [ERROR]: No Answer

FATAL ERROR: Upload Failed !!!
```

The output of the serial console is
```
OTA  : Start upload
OTA  : Progress 100%
OTA  : End

OTA  : DO NOT RESET OR POWER OFF UNTIL BOOT+FLASH IS COMPLETE.

 ets Jan  8 2013,rst cause:1, boot mode:(3,7)

load 0x4010f000, len 1384, room 16
tail 8
chksum 0x2d
csum 0x2d
v09826c6d
@cp:0
     ld
       system param error
63 6c 6f 63 6b 73 74 72 65 74 63 68 6c 69 6d 69 74 00 65 64 69 74 00 75 73 65 73 73 64 70 00 77
64 69 32 63 61 64 64 72 65 73 73 00 75 73 65 6e 74 70 00 62 61 75 64 72 61 74 65 00 73 64 6c 6f
67 6c 65 76 65 6c 00 77 65 62 6c 6f 67 6c 65 76 65 6c 00 73 65 72 69 61 6c 6c 6f 67 6c 65 76 65
6c 00 75 73 65 73 65 72 69 61 6c 00 75 64 70 70 6f 72 74 00 73 79 73 6c 6f 67 6c 65 76 65 6c 00
64 73 74 00 74 69 6d 65 7a 6f 6e 65 00 6e 74 70 68 6f 73 74 00 73 79 73 6c 6f 67 69 70 00 69 70
00 6d 65 73 73 61 67 65 64 65 6c 61 79 00 5d 7d 0a 00 0a 00 2c 00 7d 00 0a 00 2c 00 22 3a 20 00
22 00 0a 00 2c 00 22 00 22 54 61 73 6b 4e 61 6d 65 22 3a 20 22 00 7b 0a 00 22 53 65 6e 73 6f 72
73 22 3a 5b 0a 00 0a 7d 2c 0a 00 2c 0a 22 46 72 65 65 20 52 41 4d 22 3a 20 00 2c 0a 22 55 70 74
69 6d 65 22 3a 20 00 2c 0a 22 55 6e 69 74 22 3a 20 00 22 42 75 69 6c 64 22 3a 20 00 7b 22 53 79
73 74 65 6d 22 3a 7b 0a 00 74 61 73 6b 6e 72 00 55 6e 6b 6e 6f 77 6e 20 6f 72 20 72 65 73 74 72
69 63 74 65 64 20 63 6f 6d 6d 61 6e 64 21 00 65 76 65 6e 74 00 63 6d 64 00 54 6d 70 6c 4d 73 67
00 49 6e 76 61 6c 69 64 20 70 61 73 73 77 6f 72 64 21 00 3c 73 63 72 69 70 74 3e 77 69 6e 64 6f
77 2e 6c 6f 63 61 74 69 6f 6e 20 3d 20 27 2e 27 3c 2f 73 63 72 69 70 74 3e 00 3c 2f 74 61 62 6c
65 3e 3c 2f 66 6f 72 6d 3e 00 3c 54 52 3e 3c 54 44 3e 00 27 3e 3c 54 52 3e 3c 54 44 3e 3c 54 44
3e 00 3c 69 6e 70 75 74 20 74 79 70 65 3d 27 70 61 73 73 77 6f 72 64 27 20 6e 61 6d 65 3d 27 70
61 73 73 77 6f 72 64 27 20 76 61 6c 75 65 3d 27 00 3c 74 61 62 6c 65 3e 3c 54 52 3e 3c 54 44 3e
50 61 73 73 77 6f 72 64 3c 54 44 3e 00 3c 66 6f 72 6d 20 6d 65 74 68 6f 64 3d 27 70 6f 73 74 27
3e 00 70 61 73 73 77 6f 72 64 00 54 6d 70 6c 53 74 64 00 3c 2f 74 61 62 6c 65 3e 00 3c 54 52 3e
3c 54 44 3e 00 4e 6f 20 41 63 63 65 73 73 20 50 6f 69 6e 74 73 20 66 6f 75 6e 64 00 3c 74 61 62
6c 65 3e 3c 54 52 3e 3c 54 48 3e 41 63 63 65 73 73 20 50 6f 69 6e 74 73 3a 3c 54 48 3e 52 53 53
49 00 54 6d 70 6c 53 74 64 00 3c 2f 74 61 62 6c 65 3e 00 3c 54 52 3e 4e 6f 20 49 32 43 20 64 65
76 69 63 65 73 20 66 6f 75 6e 64 00 3c 54 52 3e 3c 54 44 3e 55 6e 6b 6e 6f 77 6e 20 65 72 72 6f
72 20 61 74 20 61 64 64 72 65 73 73 20 30 78 00 41 72 64 75 69 6e 6f 20 50 4d 45 00 42 4d 50 30
38 35 3c 42 52 3e 42 4d 50 31 38 30 3c 42 52 3e 42 4d 45 32 38 30 3c 42 52 3e 42 4d 50 32 38 30
3c 42 52 3e 4d 53 35 36 30 37 3c 42 52 3e 4d 53 35 36 31 31 3c 42 52 3e 48 54 31 36 4b 33 33 00
42 4d 45 32 38 30 3c 42 52 3e 42 4d 50 32 38 30 3c 42 52 3e 4d 53 35 36 30 37 3c 42 52 3e 4d 53
35 36 31 31 3c 42 52 3e 48 54 31 36 4b 33 33 00 48 54 31 36 4b 33 33 00 41 64 61 66 72 75 69 74
20 4d 6f 74 6f 72 73 68 69 65 6c 64 20 76 32 20 28 43 61 74 63 68 61 6c 6c 29 3c 42 52 3e 48 54
31 36 4b 33 33 00 41 64 61 66 72 75 69 74 20 4d 6f 74 6f 72 73 68 69 65 6c 64 20 76 32 3c 42 52
3e 53 49 31 31 34 35 00 4d 50 52 31 32 31 00 44 48 54 31 32 3c 42 52 3e 41 4d 32 33 32 30 3c 42
52 3e 42 48 31 37 35 30 3c 42 52 3e 4d 50 52 31 32 31 00 4d 50 52 31 32 31 00 4d 4c 58 39 30 36
31 34 3c 42 52 3e 4d 50 52 31 32 31 00 50 43 46 38 35 39 31 3c 42 52 3e 4d 43 50 33 32 32 31 3c
42 52 3e 4c 4d 37 35 41 00 50 43 46 38 35 39 31 3c 42 52 3e 4c 4d 37 35 41 00 50 43 46 38 35 39
31 3c 42 52 3e 41 44 53 31 31 31 35 3c 42 52 3e 54 53 4c 32 35 36 31 3c 42 52 3e 4c 4d 37 35 41
00 50 43 46 38 35 39 31 3c 42 52 3e 41 44 53 31 31 31 35 3c 42 52 3e 4c 4d 37 35 41 00 53 48 54
33 30 2f 33 31 2f 33 35 00 49 4e 41 32 31 39 00 53 49 37 30 32 31 3c 42 52 3e 48 54 55 32 31 44
3c 42 52 3e
3d 00 3e 00 20 6f 6e 63 68 61 6e 67 65 3d 22 72 65 74 75 72 6e 20 64 65 70 74 5f 6f 6e 63 68 61
6e 67 65 28 66 72 6d 73 65 6c 65 63 74 29 22 00 27 00 3c 73 65 6c 65 63 74 20 6e 61 6d 65 3d 27
00 49 32 43 20 41 64 64 72 65 73 73 00 20 2d 20 28 64 65 66 61 75 6c 74 29 00 30 78 00 00 47 50
49 4f 2d 31 36 20 28 44 30 29 00 47 50 49 4f 2d 31 35 20 28 44 38 29 00 47 50 49 4f 2d 31 34 20
28 44 35 29 00 47 50 49 4f 2d 31 33 20 28 44 37 29 00 47 50 49 4f 2d 31 32 20 28 44 36 29 00 47
50 49 4f 2d 31 31 00 47 50 49 4f 2d 31 30 20 28 44 31 32 29 00 47 50 49 4f 2d 39 20 28 44 31 31
29 00 47 50 49 4f 2d 38 00 47 50 49 4f 2d 37 00 47 50 49 4f 2d 36 00 47 50 49 4f 2d 35 20 28 44
31 29 00 47 50 49 4f 2d 34 20 28 44 32 29 00 47 50 49 4f 2d 33 20 28 44 39 29 00 47 50 49 4f 2d
32 20 28 44 34 29 00 47 50 49 4f 2d 31 20 28 44 31 30 29 00 47 50 49 4f 2d 30 20 28 44 33 29 00
2d 20 4e 6f 6e 65 20 2d 00 00 00 2d 20 4e 6f 6e 65 20 2d 00 54 6d 70 6c 53 74 64 00 44 45 42 55
47 3a 20 53 74 72 69 6e 67 20 73 69 7a 65 3a 00 68 61 6e 64 6c 65 5f 64 65 76 69 63 65 73 00 3c
2f 74 61 62 6c 65 3e 3c 2f 66 6f 72 6d 3e 00 3c 69 6e 70 75 74 20 74 79 70 65 3d 27 68 69 64 64
65 6e 27 20 6e 61 6d 65 3d 27 70 61 67 65 27 20 76 61 6c 75 65 3d 27 31 27 3e 00 3c 69 6e 70 75
74 20 74 79 70 65 3d 27 68 69 64 64 65 6e 27 20 6e 61 6d 65 3d 27 65 64 69 74 27 20 76 61 6c 75
65 3d 27 31 27 3e 00 22 3e 43 6c 6f 73 65 3c 2f 61 3e 00 3c 54 52 3e 3c 54 44 3e 3c 54 44 3e 3c
61 20 63 6c 61 73 73 3d 27 62 75 74 74 6f 6e 20 6c 69 6e 6b 27 20 68 72 65 66 3d 22 64 65 76 69
63 65 73 3f 73 65 74 70 61 67 65 3d 00 54 44 56 44 00 3c 54 44 3e 00 54 44 46 00 3c 54 44 3e 00
54 44 56 4e 00 3c 54 44 3e 00 3c 54 52 3e 3c 54 44 3e 00 3c 54 48 3e 44 65 63 69 6d 61 6c 73 00
45 61 73 79 46 6f 72 6d 75 6c 61 00 3c 54 48 3e 46 6f 72 6d 75 6c 61 00 3c 54 48 3e 4e 61 6d 65
00 3c 54 52 3e 3c 54 48 3e 56 61 6c 75 65 00 3c 2f 74 61 62 6c 65 3e 3c 74 61 62 6c 65 3e 00 56
61 6c 75 65 73 00 20 28 4f 70 74 69 6f 6e 61 6c 20 66 6f 72 20 74 68 69 73 20 44 65 76 69 63 65
29 00 73 65 63 00 54 44 54 00 44 65 6c 61 79 00 54 44 49 44 00 20 26 6e 62 73 70 3b 20 49 44 58
3a 20 00 3c 54 44 3e 00 3c 54 52 3e 3c 54 44 3e 53 65 6e 64 20 74 6f 20 43 6f 6e 74 72 6f 6c 6c
65 72 20 00 54 44 53 44 00 44 61 74 61 20 41 63 71 75 69 73 69 74 69 6f 6e 00 74 61 73 6b 64 65
76 69 63 65 70 69 6e 33 00 74 61 73 6b 64 65 76 69 63 65 70 69 6e 32 00 74 61 73 6b 64 65 76 69
63 65 70 69 6e 31 00 33 72 64 20 47 50 49 4f 00 32 6e 64 20 47 50 49 4f 00 31 73 74 20 47 50 49
4f 00 57 69 6c 6c 20 67 6f 20 69 6e 74 6f 20 65 66 66 65 63 74 20 6f 6e 20 6e 65 78 74 20 69 6e
70 75 74 20 63 68 61 6e 67 65 2e 00 54 44 50 49 00 49 6e 76 65 72 73 65 64 20 4c 6f 67 69 63 00
47 50 49 4f 2d 31 36 20 28 44 30 29 20 64 6f 65 73 20 6e 6f 74 20 73 75 70 70 6f 72 74 20 50 75
6c 6c 55 70 00 54 44 50 50 55 00 49 6e 74 65 72 6e 61 6c 20 50 75 6c 6c 55 70 00 54 44 50 00 50
6f 72 74 00 41 63 74 75 61 74 6f 72 00 53 65 6e 73 6f 72 00 54 44 47 53 00 47 6c 6f 62 61 6c 20
53 79 6e 63 00 54 44 45 00 45 6e 61 62 6c 65 64 00 54 44 4e 00 4e 61 6d 65 00 50 6c 75 67 69 6e
00 3c 54 52 3e 3c 54 44 3e 44 65 76 69 63 65 3a 3c 54 44 3e 00 54 61 73 6b 20 53 65 74 74 69 6e
67 73 00 3c 66 6f 72 6d 20 6e 61 6d 65 3d 27 66 72 6d 73 65 6c 65 63 74 27 20 6d 65 74 68 6f 64
3d 27 70 6f 73 74 27 3e 3c 74 61 62 6c 65 3e 00 3c 2f 74 61 62 6c 65 3e 3c 2f 66 6f 72 6d 3e 00
3c 54 44 3e 3c 54 44 3e 3c 54 44 3e 3c 54 44 3e 3c 54 44 3e 3c 54 44 3e ▒
▒▒▒▒x▒▒▒
▒▒▒▒
▒▒▒▒▒;▒▒x▒▒▒
▒l쎜
r▒l▒
▒▒
▒▒x▒▒▒▒▒▒▒▒
▒▒x▒▒▒l▒▒▒
▒▒▒
▒▒▒▒▒
     ▒
▒▒▒
▒▒▒▒▒▒▒▒x▒▒▒
▒▒▒▒▒
▒l▒▒▒▒
▒▒▒▒▒▒l▒▒▒▒
r▒
Exception (29):
epc1=0x4000e1c3 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000018 depc=0x00000000

ctx: cont
sp: 3fff05c0 end: 3fff0a70 offset: 01a0

>>>stack>>>
3fff0760:  4023972d 00000017 60000200 40106d24
3fff0770:  4021a9ad 402360a6 00000001 402138ce
3fff0780:  29323200 40215e97 3fff26f4 3fff219c
3fff0790:  4021abee 3fff26f4 3fff219c 00000000
3fff07a0:  3ffeee20 3ffeda70 3ffed9ec 3fff219c
3fff07b0:  00000000 00000000 00000000 00000000
3fff07c0:  00000018 00000064 80940162 fffeffff
3fff07d0:  3ff20a00 0000ffff 3fff07f8 4021653a
3fff07e0:  3ffed9ec 3fff219c 3fff219c 00000000
3fff07f0:  80940162 3fff6d79 00000000 00000000
3fff0800:  00000000 00000000 00000000 00000000
3fff0810:  00000000 00000000 00000000 00000000
3fff0820:  00000000 00000000 00000000 4021768c
3fff0830:  3fff219c 00000000 3ffeee20 402176d4
3fff0840:  00000000 3ffeda70 3ffee7f8 00000000
3fff0850:  40227095 00000003 00000003 4020323c
3fff0860:  402271e5 00000003 00000003 000e2100
3fff0870:  3ffedbe4 4022726e 00000003 00000001
3fff0880:  402075ff 3ffef7b4 00000001 00000000
3fff0890:  40207630 000e2100 3ffef7b4 40207652
3fff08a0:  3fff1388 3ffeee20 3ffeee40 40207c0f
3fff08b0:  3fff1980 00000100 0000000b 40203741
3fff08c0:  3fff1388 3fff1844 000e2100 3fff196c
3fff08d0:  3fff1388 3fff1844 000e2100 40205ab1
3fff08e0:  00000005 3fff0910 00000000 00000000
3fff08f0:  00000000 00000000 00000000 3fff1790
3fff0900:  00000071 00007100 3fff1388 40203440
3fff0910:  00000002 000000fc 00001000 63657301
3fff0920:  00000001 3fff09f0 00000074 3ffef964
3fff0930:  00000010 3fff09d4 00000000 4010053d
3fff0940:  00000100 3fff1388 3fff178c 40205698
3fff0950:  000000b1 00000000 3fff0966 3ffefa48
3fff0960:  00000010 3fff09d4 3fff09d4 4020ddfc
3fff0970:  3ffe883c 00000000 3fff09d4 4020de4b
3fff0980:  00000000 3ffef7b4 00000001 3fffdad0
3fff0990:  40227e7b 3ffef98c 3fff09d4 3ffeee20
3fff09a0:  00000001 3ffeee04 3ffef7b4 40207d90
3fff09b0:  3fffdad0 3ffeee04 3ffef7b4 40207279
3fff09c0:  3ffeee20 00000001 3fff1dc4 4020e92b
3fff09d0:  3ffeee20 3fff1dc4 0000000f 00000000
3fff09e0:  3ffeee20 000000a0 3fff0a10 4020d196
3fff09f0:  3fff1dc4 3fff1e04 00000000 00000000
3fff0a00:  3ffeee20 000000a0 000000a0 4020706d
3fff0a10:  3ffe8d90 00000000 000003e8 3ffefa48
3fff0a20:  00000003 00000000 3fff1384 4020f217
3fff0a30:  3fffdad0 3ffef98c 3ffef98c 3ffefa48
3fff0a40:  3fffdad0 3ffef98c 3ffef7b4 402073cc
3fff0a50:  feefeffe 00000000 3ffefa40 4020e5f8
3fff0a60:  feefeffe feefeffe 3ffefa50 40100718
<<<stack<<<

 ets Jan  8 2013,rst cause:1, boot mode:(3,7)

load 0x4010f000, len 1384, room 16
tail 8
chksum 0xef
csum 0xef
csum err
ets_main.c
 ets Jan  8 2013,rst cause:4, boot mode:(3,7)

wdt reset
load 0x4010f000, len 1384, room 16
tail 8
chksum 0xef
csum 0xef
csum err
ets_main.c
```

Unfortunately the intermediate image somehow doesn't connect to wifi.

Any idea why?

I can reflash it with my dev-12 firmware and start over but this is the point where I'm stuck right now. And yes I'm using the HEAD version. But it seems to be an issue with the minimal intermediate image.

Thanks for being patient.

**Comment by soif** [comment]
log seems to show that the image is crashing. It might also be due to the Spiffs settings being corrupted.
I might advise to first re uload a recent image using serial, with a prior flash erase
`espbuddy upload espeasy_1024_8285 -vbwe ` (b=vuild, w=serial, e=erase first)
then, once you have set your wifi settings in espeasy, after a hardware reboot, try to re-upload the same firmware OTA, using the intermediate step.

`espbuddy upload espeasy_1024_8285 -v `

If it crashes again, that would mean there may be an issue with intermediate firmware on ESP8265.

Thanks for the tests

PS: you can always open a second terminal session during the OTA with 👍 
`espbuddy monitor espeasy_1024_8285 -v` to see what's going on.

**Comment by lhurt** [comment]
## Next obstacle

After erasing the flash completely and flashing a release version of the ESPEasy firmware I had to enable OTA in the advanced settings.

But the firmware reboots as soon as the intermediate image is about to be uploaded.
### EspBuddy output
```
** Uploading Intermediate Uploader Firmware ****************************************************************************
-> /opt/EspBuddy/bin/espota.py -r -d -i 192.168.1.61  -f "/opt/EspBuddy/firmwares/ESPEasyUploaderMega.OTA.1m128.bin"
10:16:45 [DEBUG]: Options: {'esp_ip': '192.168.1.61', 'host_port': 30434, 'image': '/opt/EspBuddy/firmwares/ESPEasyUploaderMega.OTA.1m128.bin', 'host_ip': '0.0.0.0', 'auth': '', 'esp_port': 8266, 'spiffs': False, 'debug': True, 'progress': True}
10:16:45 [INFO]: Starting on 0.0.0.0:30434
10:16:45 [INFO]: Upload size: 288064
10:16:45 [INFO]: Sending invitation to: 192.168.1.61
10:16:45 [INFO]: Waiting for device...
10:16:55 [ERROR]: No response from device

FATAL ERROR: First Upload Failed !!!
```

### Serial console output
**Normal boot**
```
INIT : Booting version:
INIT : Warm boot #1
FS   : Mounting...
FS   : Mount successful, used 75802 bytes of 113201
CRC  : No program memory checksum found. Check output of crc2.py
CRC  : Settings CRC           ...OK
CRC  : SecuritySettings CRC   ...OK
INIT : Free RAM:20128
INIT : I2C
INIT : SPI not enabled
WIFI : AP Mode disabled
WIFI : AP Mode ssid will be ESP_Easy_0 with address 192.168.4.1
WIFI : Connecting domum nostrae attempt #1
WIFI : Connected! IP: 192.168.1.61 (ESP_Easy_0)
OTA  : Arduino OTA enabled on port 8266
WD   : Uptime 0 ConnectFailures 0 FreeMem 16336
WD   : Uptime 1 ConnectFailures 0 FreeMem 16152
```

**Start of ota**
```
OTA  : Error (will reboot): Begin Failed

 ets Jan  8 2013,rst cause:1, boot mode:(3,6)

load 0x4010f000, len 1384, room 16
tail 8
chksum 0x2d
csum 0x2d
v09826c6d
~ld
   ▒U

INIT : Booting version:
INIT : Warm boot #2
FS   : Mounting...
FS   : Mount successful, used 75802 bytes of 113201
CRC  : No program memory checksum found. Check output of crc2.py
CRC  : Settings CRC           ...OK
CRC  : SecuritySettings CRC   ...OK
INIT : Free RAM:20000
INIT : I2C
INIT : SPI not enabled
WIFI : AP Mode disabled
WIFI : AP Mode ssid will be ESP_Easy_0 with address 192.168.4.1
WIFI : Connecting domum nostrae attempt #1
WIFI : Connected! IP: 192.168.1.61 (ESP_Easy_0)
OTA  : Arduino OTA enabled on port 8266
WD   : Uptime 0 ConnectFailures 0 FreeMem 16208
WD   : Uptime 1 ConnectFailures 0 FreeMem 16304
```
May be this is an issue with the firmware or I just do something wrong. But I just compile the ESP with OTA support enabled.

My ESPeasy version was *mega-20180228*

I'm about to give up.

**Comment by soif** [comment]
Sometimes OTA fails the first time (so it reboots) and you have to insist.

Also the 'mega' branch is supposed to be large, so if you don't strip it down by removing a lot of plugins, it can simply not work, because there is not enough size for the 2 firmwares.

You have to first check that :  **firmware size + intermediate firmware size + Spiff size < 1M**

see:  
https://www.letscontrolit.com/wiki/index.php/ESPEasy#2.0.2B_firmware_on_1MB_modules

**Comment by lhurt** [comment]
I'm aware of that. My build is 605kb of size which should work.
BTW. When testing the procedure described in the article I already bricked one device.
- the upload and reboot of the intermediate image worked fine
- the upload of the new large image did not.

I don't require automation for the update but a working process (read firmware would help)
It seems to me that the ESP8285 is a little special compared to the ESP 8266 although it's claimed that it has only flash included in the package.

**Comment by soif** [comment]
**605kb is TOO MUCH !!!!**

- when you  connect (using web) to the intermediate firmware (the one linked in the wiki), it shows : **Max sketch size: 604kB**
- when you  connect (using web) to MY intermediate firmware (the one bundled with EspBuddy), it shows  : **Max sketch size: 592kB**

This **IS** your (first) issue ! 

**You MUST make tests with SMALLER firmwares** : remove most of the plugins, then build.

In my case removing the followings : 
```
_P014_SI7021.ino
_P015_TSL2561.ino
_P018_Dust.ino
_P024_MLX90614.ino
_P031_SHT1X.ino
_P032_MS5611.ino
_P039_Thermocouple.ino
_P041_NeoClock.ino
_P042_Candle.ino
_P045_MPU6050.ino
_P046_VentusW266.ino
_P047_i2c-soil-moisture-sensor.ino
_P048_Motorshield_v2.ino
_P049_MHZ19.ino
_P050_TCS34725.ino
_P051_AM2320.ino
_P052_SenseAir.ino
_P056_SDS011-Dust.ino
_P066_VEML6040.ino
_P067_HX711_Load_Cell.ino
_P068_SHT3x.ino
_P069_LM75A.ino
_P071_Kamstrup401.ino
_P072_HDC1080.ino
```
produces a 590kb firmware (on today mega branch). But you should even remove more plugins , (just to be sure that the size is small enough), as most are really not needed for your own device.

BTW I had a problem similar to yours for one of my esp8266 which hanguped during the final OTA. :bug:

I finally re-built the intermediate firmware using different compilation settings, and this fixed the issue for both of my boards. I even tried to flash with a 8285 firmware and it also worked. 🎉 🎉 

**So I suggest you to retry with the latest v1.75. It might fix your issue too !** 😸 

To be sure that the test is correct please do things in the following order.

1) be sure in the git repo to have **a smaller ESPEasy version**, then : 
`espbuddy upload espeasy_1024_8285 -vweb` (serial upload, wipe before, build first)

2) Set OTA on in the advanced settings, then reboot

3) `espbuddy upload espeasy_1024_8285 -v` (to re-upload the same firmware OTA)
Opening a serial monitor (in a second window) during this step would help to understand what is happening `espbuddy monitor espeasy_1024_8285`

If it doesn't work : Try to power off / power on the device after step 3 is completed (sometimes it helps) else retry all from scratch

**Comment by lhurt** [comment]
You're great. Thanks for your patience very much.
Now it worked. I added some addtional modules to the list and my firmware was down to 550k.
After that everything worked like a charm.

Thanks very much.

May be you can help me with two additional questions.

1. Is there a way to modify the $my_espeasy_flags variable to add a list of plugins not to compile?
2. Is it possible to modify existing Sonoffs without flashing them manually. E.g. by uploading the intermediate firmware via Web Interface, enable OTA and then do the upload of the new firmware via the espota.py command?

Thanks very much.

**Comment by soif** [comment]
🎉  Happy that you finally succeeded to use EspBuddy!!! While ALL of your issues was not **especially** related to EspBuddy itself ...

1) I can't master that myself : It's related to ESPEasy itself : ask them for a BUILD method to enable/disable plugins on built (a very USEFUL feature IMHO)

2) I'm not sure to understand what exactly is your question.....

**Comment by lhurt** [comment]
Thanks for the fast reply.
Yes I know 95% of the errors sit in front of the screen :-)

To question 2. I have a number of sonoff touch switches running vanilla ESPEasy dev-12 (without OTA). I'd like to update them and don't like the idea to remove them from the wall, hooking them up and reflashing them.
So my idea was to upload the intermediate firmware of EspBuddy via ESPEasy web interface and after reboot the should be OTA flashable. Then espota.py or a configuration with the two step process disabled of EspBuddy could do the rest.
It would be just good to get a confirmation of you if this is a feasible approach?


BTW. By disabling a ton of plugins my firmware is now 487344 bytes. This should be small enough to do the upgrade in one step. Right? Interested in the list?

**Comment by soif** [comment]
2) Yes you should be able to upload my intermediate firmware, then either update OTA, either upload using the builtin webserver. EspBuddy can **s**kip the first step just by adding the  **-s**, but it wont hurt if you'd do the 2 steps starting from the manually uploaded intermediate firmware.

>Interested in the list?

for sure

**Comment by lhurt** [comment]
Thanks.
One more thing. Is it possible to define more than one espeasy repository in espbuddy's config.php? I want to create scripted copies of the original ESPEasy firmware with different plugin configurations. As far as I understand, currently only one espeasy repository is supported.
Correct?

**Comment by soif** [comment]
yep

BTW I will soon add a 'checkout' host parameter , so that you could use different branches / host or configs , and checkout theses branches in each upload. Hence solving your issue

**Comment by lhurt** [comment]
Thanks very much.

**Comment by lhurt** [comment]
BTW this is my list of disabled plugins

```
_P003_Pulse.ino
_P004_Dallas.ino
_P005_DHT.ino
_P006_BMP085.ino
_P007_PCF8591.ino
_P008_RFID.ino
_P009_MCP.ino
_P010_BH1750.ino
_P011_PME.ino
_P012_LCD.ino
_P013_HCSR04.ino
_P014_SI7021.ino
_P015_TSL2561.ino
_P016_IR.ino
_P017_PN532.ino
_P018_Dust.ino
_P019_PCF8574.ino
_P021_Level.ino
_P022_PCA9685.ino
_P023_OLED.ino
_P024_MLX90614.ino
_P025_ADS1115.ino
_P027_INA219.ino
_P028_BME280.ino
_P029_Output.ino
_P030_BMP280.ino
_P031_SHT1X.ino
_P032_MS5611.ino
_P034_DHT12.ino
_P035_IRTX.ino
_P036_FrameOLED.ino
_P038_NeoPixel.ino
_P039_Thermocouple.ino
_P040_ID12.ino
_P041_NeoClock.ino
_P042_Candle.ino
_P044_P1WifiGateway.ino
_P045_MPU6050.ino
_P046_VentusW266.ino
_P047_i2c-soil-moisture-sensor.ino
_P048_Motorshield_v2.ino
_P049_MHZ19.ino
_P050_TCS34725.ino
_P051_AM2320.ino
_P052_SenseAir.ino
_P053_PMSx003.ino
_P054_DMX512.ino
_P056_SDS011-Dust.ino
_P057_HT16K33_LED.ino
_P058_HT16K33_KeyPad.ino
_P059_Encoder.ino
_P060_MCP3221.ino
_P061_KeyPad.ino
_P062_MPR121_KeyPad.ino
_P063_TTP229_KeyPad.ino
_P064_APDS9960.ino
_P065_DRF0299_MP3.ino
_P066_VEML6040.ino
_P067_HX711_Load_Cell.ino
_P068_SHT3x.ino
_P069_LM75A.ino
_P070_NeoPixel_Clock.ino
_P071_Kamstrup401.ino
_P072_HDC1080.ino
_P073_7DGT.ino
_P074_TSL2591.ino
```

**Comment by soif** [comment]
thanks 🎉

**Comment by Oxyandy** [comment]
> BTW in order to get OTA working, your current running firmware MUST have be compiled with the FEATURE_ARDUINO_OTA defined, ie:
> 
> $my_espeasy_flags=''-DFEATURE_ARDUINO_OTA; or de-commented in the ESPEASY.ino file.

Not sure about above comment ^^, I fixed OTA on my Sonoff Basic devices, using an initial uploader and modified firmware.
I did not need to recompile anything to make it work such as - enable OTA
It works really, really well - over 99% of the time, first time it just works..
The 1% just needs a second try to work. ;)
I can flash my Sonoff from R78 - R148 to v2.0.0-dev4 - v2.0.0-dev13 & even latest Mega releases 
(although very latest maybe now getting too large)
However I did modify all firmwares to ensure they worked correctly..
And... I do use the devices web-page...
![ota](https://user-images.githubusercontent.com/36609061/38463233-250b2e0c-3b39-11e8-9c78-12707ac3d511.jpg)

Do you mean you need to 'enable ota' to 'push' the update via http ?
Cheers

**Comment by soif** [comment]
> Do you mean you need to 'enable ota' to 'push' the update via http ?

No as stated this was needed to get OTA working.

**Comment by Oxyandy** [comment]
Ok curious, for me with my 1M Sonoff Basics, with DOUT uploader & modified original releases, it works perfect through the browser "Tools -> Firmware -> Load" without recompiling.
Thanks for your response, still learning
